### DISCUSSION TITLE
Limitations in Platform API

### DISCUSSION THREAD

**Discussion by shykes** [dagger/dagger#12164] [Ideas]
## Problem

The Dagger API lets you manage the platform (OS + Architecture) of containers. But there are limitations in the API that make some use cases awkward, or sometimes impossible.

For example, given a `Container` that may point to a multi-architecture image, I can't select a platform:

```golang
// Return the version of an image for a specific platform, or an error if the platform is not available
func GetPlatform(img *dagger.Container, platform dagger.Platform) (*dagger.Container, error) {
  // ???
}
```

Also, publishing is awkward. (FIXME: example needed).

## Solution

Improve Dagger API to make platform management less awkward.

**Comment by sagikazarmark** [comment]
Related https://github.com/dagger/dagger/issues/6675
